### PR TITLE
Modernization

### PR DESCRIPTION
**Pull request by BartMassey**
This cleans up and modernizes the code a bit, adds rustdoc, and fixes Linux to build and run correctly.

The new Windows and Mac code is completely untried.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,40 +1,100 @@
...
+    use super::*;
+    use winreg::enums::HKEY_LOCAL_MACHINE;
+
+    /// Get the hardware ID of this machine. The HWID is
+    /// obtained from the Windows registry at location
+    /// `\\\\SOFTWARE\\Microsoft\\Cryptography\\MachineGuid`.
+    pub fn get_id() -> Result<std::string::String, HwIdError> {
+        // escaping is fun, right? right???
+        let hive = winreg::RegKey::predef(HKEY_LOCAL_MACHINE)
+            .open_subkey(r"\\SOFTWARE\Microsoft\Cryptography")
```
Comment (tilda) [inline src/lib.rs]: change the argument for `open_subkey` to this: `r"Software\Microsoft\Cryptography"`

apparently despite the key being called "SOFTWARE" in regedit you have to refer to it as "Software". classic windows I guess 🤷.

**Comment by tilda** [comment]
on the windows side seems to compile & run example just fine with the change I pointed out in review. appreciate the contribution!

later edit: i'm also unsure on if it works on macos - i don't exactly have one lying around and don't feel like getting a vm up & running just for that. i'll see if i can work with github actions for that after this is merged probably

File: Cargo.toml [hunk 2]
```diff
@@ -1,18 +1,13 @@
...
-[dependencies]
+# https://stackoverflow.com/a/51821632/364875
+[target.'cfg(target_os = "windows")'.dependencies]
 winreg = "0.6"
 
-[lib]
-name = "hwid"
-path = "src/lib.rs"
-
-[[bin]]
```
Comment (tilda) [inline Cargo.toml:16]: could add it back:
```toml
[[example]]
name = "get_hwid"
crate-type = ["bin"]
```

Comment (BartMassey) [reply Cargo.toml:16]: Up to you. I think the simpler config is better, but I'm fine either way.

File: src/lib.rs (hunk 1 above)
Comment (BartMassey) [reply src/lib.rs]: Nice. :shrug:  I guess it's one of those Windows "capitalize eight letters or less" things.

Thanks much, change pushed!

File: Cargo.toml (hunk 2 above)
Comment (tilda) [reply Cargo.toml:16]: Fair enough - won't force it in this PR at least

**Comment by BartMassey** [comment]
Thanks for the merge!

**Pull request merged**